### PR TITLE
 feat(api) : event stream producer added 

### PR DESCRIPTION
**Pull request by soumyadip007**
Subject:  feat(api) : event stream producer added 
Assignees: @soumyadip007 

## Closes / Fixes / Resolves

1. Deployment connection and analytics typo fixed
2. SSE producer implemented 
3. Events added into API
4. SSE added for all the activities 
5. /analytics/events end point added for SSE
6. Realtime Activity Stream Creation, Deletion of Env, Property, APIKey, Deployment of Applications
7. Service Documentation added for the services

## Does this PR introduce a breaking change

NO

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api/src/app.module.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ import { ApikeyModule } from './server/api-key/service/apikey.module';
 import { ApplicationController } from './server/application/application.controller';
 import { ApplicationModule } from './server/application/service/application.module';
 import { AuthModule } from './server/auth/auth.module';
-import { DeploymentConnectionController } from './server/deployment-conncetion/deployment-connection.controller';
-import { DeploymentConnectionModule } from './server/deployment-conncetion/service/deployment-connection.module';
+import { DeploymentConnectionController } from './server/deployment-conncection/deployment-connection.controller';
+import { DeploymentConnectionModule } from './server/deployment-conncection/service/deployment-connection.module';
```
Comment (hybridx) [inline packages/api/src/app.module.ts:10]:
```suggestion
import { DeploymentConnectionController } from './server/deployment-conncection/deployment-connection.controller';
import { DeploymentConnectionModule } from './server/deployment-conncection/service/deployment-connection.module';
```

File: packages/api/src/repository/data-services.abstract.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { ActivityStream } from 'src/server/analytics/activity-stream.entity';
 import { Apikey } from '../server/api-key/apikey.entity';
 import { Application } from '../server/application/application.entity';
-import { DeploymentConnection } from '../server/deployment-conncetion/deployment-connection.entity';
+import { DeploymentConnection } from '../server/deployment-conncection/deployment-connection.entity';
```
Comment (hybridx) [inline packages/api/src/repository/data-services.abstract.ts:4]:
```suggestion
import { DeploymentConnection } from '../server/deployment-conncection/deployment-connection.entity';
```

File: packages/api/src/server/analytics/service/analytics.service.ts [hunk 3]
```diff
@@ -76,4 +84,12 @@ export class AnalyticsService {
...
-}
+
+  subscribe() {
+    return fromEvent(AnalyticsService.emitter, AnalyticsService.channel);
+  }
+
+  emit(data) {
+    AnalyticsService.emitter.emit(AnalyticsService.channel, data);
+  }
+}
```
Comment (hybridx) [inline packages/api/src/server/analytics/service/analytics.service.ts:95]:
```suggestion
}

```

File: packages/api/src/server/application/service/application.service.ts [hunk 4]
```diff
@@ -88,7 +88,7 @@ export class ApplicationService {
         propertyIdentifier,
         env
       });
-      console.log(agendaResponse);
+      this.logger.log('Agenda', JSON.stringify(agendaResponse));
```
Comment (hybridx) [inline packages/api/src/server/application/service/application.service.ts:91]: Do we need the entire response?
We can add this response into different levels right?

File: packages/api/src/server/deployment-conncection/service/deployment-connection.factory.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { Injectable } from '@nestjs/common';
 import { DeploymentConnection } from 'src/repository/mongo/model';
-import { DeploymentConnectionDTO, UpdateDeploymentConnectionDTO } from 'src/server/deployment-conncetion/deployment-connection.dto';
+import { DeploymentConnectionDTO, UpdateDeploymentConnectionDTO } from 'src/server/deployment-conncection/deployment-connection.dto';
```
Comment (hybridx) [inline packages/api/src/server/deployment-conncection/service/deployment-connection.factory.ts:3]:
```suggestion
import { DeploymentConnectionDTO, UpdateDeploymentConnectionDTO } from 'src/server/deployment-conncection/deployment-connection.dto';
```

File: packages/api/src/server/deployment-conncection/service/deployment-connection.service.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { Injectable } from '@nestjs/common';
 import { IDataServices } from 'src/repository/data-services.abstract';
 import { DeploymentConnection } from 'src/repository/mongo/model';
-import { DeploymentConnectionDTO, UpdateDeploymentConnectionDTO } from 'src/server/deployment-conncetion/deployment-connection.dto';
+import { DeploymentConnectionDTO, UpdateDeploymentConnectionDTO } from 'src/server/deployment-conncection/deployment-connection.dto';
```
Comment (hybridx) [inline packages/api/src/server/deployment-conncection/service/deployment-connection.service.ts:4]:
```suggestion
import { DeploymentConnectionDTO, UpdateDeploymentConnectionDTO } from 'src/server/deployment-connection/deployment-connection.dto';
```

**Review by hybridx — CHANGES_REQUESTED** [review]

File: packages/api/src/server/deployment-conncection/service/deployment-connection.service.ts (hunk 6 above)
Comment (soumyadip007) [reply packages/api/src/server/deployment-conncection/service/deployment-connection.service.ts]: Resolved

File: packages/api/src/repository/data-services.abstract.ts (hunk 2 above)
Comment (soumyadip007) [reply packages/api/src/repository/data-services.abstract.ts]: Resolved

File: packages/api/src/server/analytics/service/analytics.service.ts (hunk 3 above)
Comment (soumyadip007) [reply packages/api/src/server/analytics/service/analytics.service.ts]: Resolved

**Comment by soumyadip007** [comment]
@hybridx  I've resolved your comments and added the documentation for the services. Please check.

**Review by hybridx — APPROVED** [review]

**Pull request merged**